### PR TITLE
Fixed several missing links in docs.

### PR DESCRIPTION
**Pull request by Wcubed**
Links in the api docs are nice. I noticed that there were several places where structs / functions and other things were referenced in the docs, but weren't linked. I added the links where possible / logical.

### CODE REVIEW

File: crates/bevy_render/src/texture/image.rs [hunk 1]
```diff
@@ -151,7 +151,7 @@ impl ImageSampler {
         }
     }
 
-    /// Returns a sampler descriptor with `Nearest` min and mag filters
+    /// Returns a sampler descriptor with [`Linear`](crate::render_resource::FilterMode) min and mag filters
```
[the remark this answers is not in the document]
Comment (Wcubed) [reply crates/bevy_render/src/texture/image.rs]: Good catch. Fixed 👍

**Comment by james7132** [comment]
@Wcubed can you resolve these conflicts?

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_app/src/app.rs [hunk 2]
```diff
@@ -672,7 +672,7 @@ impl App {
         }
     }
 
-    /// Boxed variant of `add_plugin`, can be used from a [`PluginGroup`]
+    /// Boxed variant of [`add_plugin`](App::add_plugin) can be used from a [`PluginGroup`]
```
Comment (mockersf) [inline crates/bevy_app/src/app.rs:675]:
```suggestion
    /// Boxed variant of [`add_plugin`](App::add_plugin) that can be used from a [`PluginGroup`]
```

Comment (mockersf) [reply crates/bevy_app/src/app.rs:675]: or keeping the original comma, but I prefer words 🤷

File: examples/audio/decodable.rs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//! Shows how to create a custom [`Decodable`] type by implementing a Sine wave.
 //! Shows how to create a custom `Decodable` type by implementing a Sine wave.
```
Comment (mockersf) [inline examples/audio/decodable.rs:2]:
```suggestion
//! Shows how to create a custom [`Decodable`] type by implementing a Sine wave.
```

**Review by hymm — APPROVED** [review]
I just did a quick sanity check review. Trusting CI to know if the links are actually correct or not.

**Review by mockersf — APPROVED** [review]

**Pull request merged**